### PR TITLE
feat: add billing information support

### LINKED ISSUE #107: Add optional billing info step to the payment widget ("Add Step 5")
# Problem

Per [Payment Template Designs Miro](https://miro.com/app/board/uXjVK_t9NEA=/), payment widget is missing step 5

# Definiton of Done

- [ ] If the Seller provides seller info, it should include the invoice number. (Copied from #112)

### LINKED ISSUE #110: [payment-widget] Change rn-checkout to "receipt" if no billing info is added

### PR DESCRIPTION
**Pull request by aimensahnoun**
Resolves #107 
Resolves #110 

- Adds props for `sellerInfo`, `buyerInfo`, `invoiceNumber` , and `enableBuyerInfo`
- Shows a new step for buyer to fill in their billing information if `enableBuyerInfo` is set to true (default to true)
- Skips the billing information step if `enableBuyerInfo` is set to false
- Follows the invoice format

CleanShot 2024-09-03 at 22 52 14

CleanShot 2024-09-03 at 22 51 11

## Summary by CodeRabbit

- **New Features**
	- Introduced a buyer information form for collecting essential buyer details within the payment widget.
	- Enhanced the currency selector with improved user interaction and modularity.
	- Updated payment confirmation to handle detailed seller, buyer, and product information.
	- Added support for dynamic payment steps based on buyer information presence.

- **Bug Fixes**
	- Improved validation logic for buyer information to ensure data integrity before proceeding.

- **Documentation**
	- Updated type definitions and interface documentation to reflect new properties and structures for better clarity.

- **Chores**
	- Refined internal request handling to accommodate new data structures for improved transaction management.

### CODE REVIEW

**Comment by MantisClone** [comment]
Just a head's up @aimensahnoun you have to do the magic word for both PRs for it to link to both of them automatically
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
Screenshot from 2024-09-03 19-40-06

File: packages/payment-widget/src/lib/types/index.ts [hunk 1]
```diff
@@ -1,9 +1,36 @@
 import type { CURRENCY_ID, NETWORK_LABEL } from "../utils/currencies";
 
-export type SellerInfo = {
+export interface Address {
+  "street-address"?: string;
+  locality?: string;
+  region?: string;
+  "country-name"?: string;
+  "postal-code"?: string;
+}
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply packages/payment-widget/src/lib/types/index.ts:9]: Thanks for the suggestion. The address is validated by the Request Network SDK so I don't think need to validate it here.

File: packages/payment-widget/src/lib/types/index.ts [hunk 2]
```diff
@@ -27,4 +54,8 @@ export type Currency = {
   name?: string;
 };
 
-export type PaymentStep = "currency" | "confirmation" | "complete";
+export type PaymentStep =
+  | "currency"
+  | "buyer-info"
+  | "confirmation"
+  | "complete";
```
[the remark this answers is not in the document]
Comment (MantisClone) [reply packages/payment-widget/src/lib/types/index.ts:61]: This will be added in a future PR.

**Review by MantisClone — APPROVED** [review]

File: packages/payment-widget/src/lib/types/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (aimensahnoun) [reply packages/payment-widget/src/lib/types/index.ts:61]: `buyer-info` is used in `payment-widget` and `payment-confirmation` components to navigate between steps.

**Pull request merged**